protocol/fusaka-on-l2.md

```diff
@@ -0,0 +1,101 @@
+
+## Problem Statement + Context
+
+<!-- Describe the specific problem that the document is seeking to address as well
+as information needed to understand the problem and design space.
+If more information is needed on the costs of the problem,
+this is a good place to that information. -->
+
+The following EIPs were added in the L1 Fusaka hardfork but are not relevant to L2.
+
+| EIP | Why irrelevant? |
+|-----|-----------------|
+| [7594 PeerDAS](https://eips.ethereum.org/EIPS/eip-7594) | Blob optimization |
+| [7892 BPO Forks](https://eips.ethereum.org/EIPS/eip-7892) | Blob optimization |
+| [7917 Deterministic Proposer Lookahead](https://eips.ethereum.org/EIPS/eip-7917) | L1 CL detail |
+| [7918 Bounded Blob Base Fee](https://eips.ethereum.org/EIPS/eip-7918) | Blob optimization |
+| [7935 Raise Default Gas Limit](https://eips.ethereum.org/EIPS/eip-7935) | OP Stack operator can set the block gas limit |
+| [7934 RLP EL Block Size Limit](https://eips.ethereum.org/EIPS/eip-7934) | Sequencer policy is sufficient |
+
+The following are the Fusaka EIPs that can be adopted on L2.
+
+| EIP |  Adoption Questions  |
+|-----|----------------------|
+| [7642 eth/69](https://eips.ethereum.org/EIPS/eip-7642)\* | |
+| [7823 Upper-Bound MODEXP](https://eips.ethereum.org/EIPS/eip-7823) | |
+| [7825 Tx Gas Limit Cap](https://eips.ethereum.org/EIPS/eip-7825) |  What should the cap be on L2? Should it be configurable? |
+| [7883 MODEXP Gas Cost Increase](https://eips.ethereum.org/EIPS/eip-7883) | |
+| [7910 eth_config Endpoint](https://eips.ethereum.org/EIPS/eip-7910)\* | |
+| [7939 CLZ Opcode](https://eips.ethereum.org/EIPS/eip-7939) | |
+| [7951 secp256r1 Precompile](https://eips.ethereum.org/EIPS/eip-7951) | Already adopted: should L2s use the new gas cost? |
+
+\* Can technically be implemented outside of a hardfork, but we intend to ship them as part of a hardfork since they require multi-client compatibility.
+
+We need to decide the following:
+
+* Among the Relevant EIPs:
+  * Answer the adoption questions.
+  * Are there any other adoption concerns associated with the listed EIPs?
+* Among the Irrelevant EIPs, are there any relevant EIPs that should be included on L2?
+
+## Proposed Solution
+
+<!-- A high level overview of the proposed solution.
+When there are multiple alternatives there should be an explanation
+of why one solution was picked over other solutions.
+As a rule of thumb, including code snippets (except for defining an external API)
+is likely too low level. -->
+
+Assuming the division of relevant and irrelevant EIPs is correct, we need to answer the adoption questions.
+
+### 7825 Tx Gas Limit Cap: What should the cap be on L2? Should it be configurable?
+
+We propose adopting a constant cap that matches L1: 16,777,216 gas (2^24).
+This is the simplest option that maximizes Ethereum equivalence.
+If there is a strong need for a different cap or flexibility in setting the cap, that can be considered in future hardforks.
```

Comment:
I agree. There's a lot of value in maximizing Eth equivalence.

The one exception I propose to make here is for deposit transactions. These should be excluded from the gas limit. This makes it easier to any future crazy upgrade txs to not need to work around such a limit, like the L2CM migration.

Reply:
I feel this is also somewhat related to https://github.com/ethereum-optimism/design-docs/pull/351#discussion_r2498337596

Can we add a way for the EL to check that a deposit tx is an upgrade transaction, and not subtract the upgrade transaction's gas used from the block gas limit? That way inclusion is guaranteed.

Reply:
We agree too. We've already been [enforcing a 25M cap at the mempool level](https://docs.base.org/base-chain/network-information/block-building#per-transaction-gas-maximum) since September 17, and we want to further reduce that to 16.7M along with the rest of the Superchain when Fusaka's EIPs make it into OP Stack. This lower limit improves flashblock performance and mitigates the degradation than underpriced opcodes can cause (e.g. XEN transactions).